### PR TITLE
Added script for metrics module with refresh metrics custom function

### PR DESCRIPTION
**Pull request by vijay4953**
### Requirements

To verify the metrics module is working fine in Jazz application and also verify count is reflecting under metrics tab.

### Benefits

We can easily verify the metrics module by checking the count increase through automation. No manual intervention required.

### Possible Drawbacks

Initially, We can't be able to test the count increase for lambda and website because application is not giving the proper result for now.

### Applicable Issues

 count increase for lambda and website because application is not giving the proper result for now.

### CODE REVIEW

**Comment by pspani** [comment]
please fix merge conflicts

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+
+  it('Create API Service', function () {
+    browser.driver.sleep(fivek);
+    browser.wait(EC.visibilityOf(jazzServices_po.getCreateService()), timeOutHigh);
+    browser.wait(EC.elementToBeClickable(jazzServices_po.getCreateService()), timeOutHigh);
+    winhandle = browser.getWindowHandle();
+    //To create Service-API
+    jazzServices_po.getCreateService().click();
+    var min = 111111111;
```
Comment (pspani) [inline core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: shift to a configuration file

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    serviceapprover();
+    browser.driver.sleep(fifteenk);
+    //Assert-Verifying the created service,Type and Status of the API
+    expect(jazzServices_po.getService(servicename).getText()).toEqual(servicename);
+    fluentwaittry(jazzServices_po.getAPIType(servicename), tenk);
+    expect(jazzServices_po.getAPIType(servicename).getText()).toEqual('api');
+    expect(jazzServices_po.getAPIStatus(servicename).getText()).toEqual('creation started');
+    waitforservice(jazzServices_po.serviceStatus(servicename), sixtyk);
+    expect(jazzServices_po.getAPIStatus(servicename).getText()).toEqual('active');
+    console.log("service created");
```
Comment (pspani) [inline core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: remove console log statements

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+          },
+          function (error) {
+            browser.refresh();
+            return false;
+          });
+    }, 240 * 1000);
+  }
+
+  function refreshbutton(ele, t) {
```
Comment (pspani) [inline core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: frequently used functions should be moved to a util file

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    browser.manage().timeouts().implicitlyWait(0);
+//     if (browser.wait(EC.elementToBeClickable(jazzServices_po.getRefresh()), fivek))
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: addressed

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+    });
+    console.log("metrics navigation done");
+  });
+
+  it('Verify API Deployments', () => {
+    browser.driver.sleep(twok);
+    refreshbutton(jazzServices_po.getDeploymentStatus(), fivek);
+    jazzServices_po.getDeploymentStatus().click();
+    waitForSpinnerDisappear();
+    refreshbutton(jazzServices_po.getDeploymentStatusVerify(), fivek);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: no this is right

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+    expect(jazzServices_po.getDeploymentStatusVerify().getText()).toEqual('Successful');
+  });
+
+  it('Verify API Asset', () => {
+    browser.driver.sleep(twok);
+    refreshbutton(jazzServices_po.getAsset(), fivek);
+    //To get the Asset Tab
+    jazzServices_po.getAsset().click();
+    waitForSpinnerDisappear();
+    refreshbutton(jazzServices_po.getAssetStatusVerify(), fivek);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: no

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    jazzServices_po.getService(servicename).click();
+    fluentwaittry(jazzServices_po.getServiceNameHeader(), fivek);
+    fluentwaittry(jazzServices_po.getProdName(), fivek);
+    jazzServices_po.getProdName().click();
+    waitForSpinnerDisappear();
+    browser.refresh();
+    fluentwaittry(jazzServices_po.getMetrices(), tenk);
+    jazzServices_po.getMetrices().click();
+    waitForMetricsSpinner();
+    refreshbutton(jazzServices_po.getMetricesCount(), thirtyk);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: no

File: core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+    refreshbutton(jazzServices_po.getMetrices(), fivek);
+    jazzServices_po.getMetrices().click();
+    waitForSpinnerDisappear();
+  });
+
+  it('Verify Lambda Deployments', () => {
+    refreshbutton(jazzServices_po.getDeploymentStatus(), fivek);
+    jazzServices_po.getDeploymentStatus().click();
+    waitForSpinnerDisappear();
+    refreshbutton(jazzServices_po.getDeploymentStatusVerify(), fivek);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts]: no

File: core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+ * limitations under the License.
+ */
+
+import { browser, element, by, protractor, $ } from 'protractor';
+import { Jazz } from '../page-objects/jazzservices.po';
+import { CONFIGURATIONS } from '../../src/config/configuration';
+import { Timeouts, Browser } from 'selenium-webdriver';
+import { SSL_OP_SSLEAY_080_CLIENT_DH_BUG } from 'constants';
+
+const twok = 2000;
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts]: done

File: core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+
+  it('Verify Lambda Asset', () => {
+    refreshbutton(jazzServices_po.getAsset(), fivek);
+    //To get the Asset Tab
+    jazzServices_po.getAsset().click();
+    waitForSpinnerDisappear();
+    refreshbutton(jazzServices_po.getAssetStatusVerify(), fivek);
+    //Verifying the Assets are ACTIVE
+    expect(jazzServices_po.getAssetStatusVerify().getText()).toEqual('ACTIVE');
+    refreshbutton(jazzServices_po.getAssetHeader(), fivek);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/05-lambdaservices/lambdaservices.e2e-spec.ts]: This function verify the header of asset

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+      return ele.isDisplayed()
+        .then(
+          function (text) {
+            return text;
+          },
+          function (error) {
+            jazzServices_po.getRefresh().click();
+            return false;
+          });
+    }, 120 * 1000);
```
[the remark this answers is not in the document]
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: moved this function to the common function and reduced the time also

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts (hunk 1 above)
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: kept in that files and tried it was not working

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts (hunk 2 above)
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: removed

File: core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts (hunk 3 above)
Comment (vijay4953) [reply core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts]: removed

**Pull request merged**